terraform/main.tf

```diff
@@ -0,0 +1,194 @@
+provider "aws" {
+  region = "us-east-1"
+}
+
+data "archive_file" "app_zip" {
+  type        = "zip"
+  source_dir  = "../app/dist"
+  output_path = "./app.zip"
+}
+
+resource "aws_apigatewayv2_api" "app" {
+  name          = "api"
+  protocol_type = "HTTP"
+}
+
+resource "aws_apigatewayv2_stage" "app" {
+  api_id = aws_apigatewayv2_api.app.id
+
+  name        = "api"
+  auto_deploy = true
+
+  access_log_settings {
+    destination_arn = aws_cloudwatch_log_group.api_gw.arn
+
+    format = jsonencode({
+      requestId               = "$context.requestId"
+      sourceIp                = "$context.identity.sourceIp"
+      requestTime             = "$context.requestTime"
+      protocol                = "$context.protocol"
+      httpMethod              = "$context.httpMethod"
+      resourcePath            = "$context.resourcePath"
+      routeKey                = "$context.routeKey"
+      status                  = "$context.status"
+      responseLength          = "$context.responseLength"
+      integrationErrorMessage = "$context.integrationErrorMessage"
+    }
+    )
+  }
+}
+
+resource "aws_apigatewayv2_integration" "app" {
+  api_id = aws_apigatewayv2_api.app.id
+
+  integration_uri    = aws_lambda_function.app.invoke_arn
+  integration_type   = "AWS_PROXY"
+  integration_method = "POST"
+}
+
+resource "aws_apigatewayv2_route" "app" {
```

Comment:
Imagine you had to protect all your routes with a common authentication and authorization logic. 

How would you structure that?

Reply:
Need to add middleware to check auth and pass to context current user